### PR TITLE
Feature: Propose new settings page

### PR DESCRIPTION
**Pull request by RakeshUP**
## Description

TODO:
- Execution Card needs to be changed to factor in new/old settings as action
- Edit button in description list is dummy for now. maybe we need to use a single formcontext over both edit settings and propose settings page?

Task: [426](https://aragonassociation.atlassian.net/jira/software/c/projects/APP/boards/41?modal=detail&selectedIssue=APP-426)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/pages/proposeSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    mode: 'onChange',
+  });
+
+  return (
+    <>
+      <FormProvider {...formMethods}>
+        <FullScreenStepper
+          wizardProcessName={t('newProposal.title')}
+          navLabel={t('navLinks.settings')}
+          returnPath={generatePath(Settings, {network, dao})}
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposeSettings.tsx]: Are you sure this shouldn't take you to the edit settings page rather than just the settings page?

**Review by WissenIstNacht — COMMENTED** [review]
Thanks for the PR, looks good!

Regarding your comments on this PR: Which "edit" button do you mean, exactly? At the moment the description cards (DAO Metadata/Governance) on the Propose new Settings step (1/4), do not have  a button. Did you mean something esle? Or did you forget to add them? 

As for your question on how to keep track of things: A while back there was the idea to have the values entered by a user on a form should be handled withing the use-react-form library, while other  state (more meta infos about the form flow) were to be handled in a context wrapped around that flow. No idea if this was ever done or not though. :D

**Review by Fabricevladimir — APPROVED** [review]
Apart from @WissenIstNacht's comment, looks good to me! Thanks

File: packages/web-app/src/pages/proposeSettings.tsx (hunk 1 above)
Comment (RakeshUP) [reply packages/web-app/src/pages/proposeSettings.tsx]: Fixed

**Comment by RakeshUP** [comment]
> Regarding your comments on this PR: Which "edit" button do you mean, exactly? At the moment the description cards (DAO Metadata/Governance) on the Propose new Settings step (1/4), do not have a button. Did you mean something esle? Or did you forget to add them?

Yes, I forgot to add the "Edit Settings" button on the description list. Added that now! 😅
> As for your question on how to keep track of things: A while back there was the idea to have the values entered by a user on a form should be handled withing the use-react-form library, while other state (more meta infos about the form flow) were to be handled in a context wrapped around that flow. No idea if this was ever done or not though. :D

In this particular case, we need the form data to be shared and persisted between URL changes for both edit settings page and propose settings page. So a single page form context won't work here and we have to come up with a solution for this

**Review by sepehr2github — COMMENTED** [review]
Looks good to me, just a minor thing and don't know if It's sth related to this PR or not, but the resource card look broken when we didn't pass any resources.

Screen Shot 2022-06-02 at 11 00 03 AM

**Comment by RakeshUP** [comment]
@sepehr2github yes that's a known issue, the resource list has no empty state in design and we'll have to create a design ticket for that.

**Review by sepehr2github — APPROVED** [review]
Thank you for the changes, and tested, good to go!

**Comment by sepehr2github** [comment]
> @sepehr2github yes that's a known issue, the resource list has no empty state in design and we'll have to create a design ticket for that.

Ok, May I ask you to create a bug ticket on jira so we can don't forget to fix it in the future.

**Pull request merged**